### DISCUSSION TITLE
KML import fails, error logs?

### DISCUSSION THREAD

**Discussion by palzkillm** [organicmaps/organicmaps#6463] [Q&A]
Hey,
I'm used to import KML files into Organic Maps. I write those KMLs myself with an App I coded myself. All of a sudden, OM stopped importing them, it says, they have some error. This could be due to me working on the app generating the files, but as far as I can see, I did not change anything regarding the KML output.

Can someone help me find an actual error log, which helps me debug the files? 

One of them is attached for reference.
[MPA_2023-11_Las_Palmas_20231103.zip](https://github.com/organicmaps/organicmaps/files/13253882/MPA_2023-11_Las_Palmas_20231103.zip)

Marco

**Comment by biodranik** [comment]
Check the log output of Organic Maps, it should tell you the line number with the error. You likely have an invalid XML or some invalid, non-XML symbol in it.

**Reply by palzkillm** [reply]
And where/how could I find that? Using the iOS App.

**Reply by biodranik** [reply]
You may try [this instruction](https://www.hexnode.com/mobile-device-management/help/obtain-ios-device-logs-using-mac-and-windows/), or try to build/run a desktop OM app on Mac or Linux.

**Comment by cyber-toad** [comment]
@biodranik I tested this file with android beta and you were correct about symbol, the error is:
```
runtime.cc:638] JNI DETECTED ERROR IN APPLICATION: input is not valid Modified UTF-8: illegal continuation byte 0x61
untime.cc:638]     string: 'kml/serdes.hpp:163 Deserialize(): <?xml version="1.0" encoding="UTF-8"?><kml xmlns="http://www.opengis.net/kml/2.2"><Document><name>2023-11_Las_Palmas_20231103</name><Style id="placemark-blue"><IconStyle><color>ffff0000</color><Icon><href>https://www.gstatic.com/mapspro/images/stock/503-wht-blank_maps.png</href></Icon></IconStyle></Style><Style id="placemark-green"><IconStyle><color>ff00ff00</color><Icon><href>https://www.gstatic.com/mapspro/images/stock/503-wht-blank_maps.png</href></Icon></IconStyle></Style><Style id="placemark-brown"><IconStyle><color>ff00458B</color><Icon><href>https://www.gstatic.com/mapspro/images/stock/503-wht-blank_maps.png</href></Icon></IconStyle></Style><Style id="placemark-orange"><IconStyle><color>ff008cff</color><Icon><href>https://www.gstatic.com/mapspro/images/stock/503-wht-blank_maps.png</href></Icon></IconStyle></Style><Folder><name>2023-11_Las_Palmas_20231103</name><Placemark><name><![CDATA[LF - Canary Crusader LAB : S1 Las Palmas de Gran Canar
```

is it something we can quickly fix on our side or xml parser upgrade/etc is required? I was able to load this kml to Google Earth.

**Reply by biodranik** [reply]
Incorrectly created KML should be fixed to follow XML/KML standard. There is a bug in _another code_ that should be fixed.
